packages/core/src/discovery.js

```diff
@@ -218,7 +218,7 @@ async function* captureSnapshotResources(page, snapshot, options) {
   const log = logger('core:discovery');
   let { discovery, additionalSnapshots = [], ...baseSnapshot } = snapshot;
   let { capture, captureWidths, deviceScaleFactor, mobile, captureForDevices } = options;
-  let cookies = snapshot?.domSnapshot?.cookies || snapshot?.domSnapshot?.[0]?.cookies;
+  let cookies = snapshot.domSnapshot?.cookies || snapshot.domSnapshot?.[0]?.cookies;
```

Comment:
```suggestion
  let cookies = snapshot.domSnapshot?.cookies || (Array.isArray(snapshot.domSnapshot) && snapshot.domSnapshot[0]?.cookies);
```

Reply:
Are you recommending it for change or style ?

Reply:
no looks good